Content.Server/Explosion/EntitySystems/ExplosionSystem.Processing.cs

```diff
@@ -463,9 +470,44 @@ private void ProcessEntity(
 
                 }
 
-                // TODO EXPLOSIONS turn explosions into entities, and pass the the entity in as the damage origin.
-                _damageableSystem.TryChangeDamage(entity, damage * _damageableSystem.UniversalExplosionDamageModifier, ignoreResistances: true);
+                if (type.BlacklistedTags != null)
+                {
+                    bool skipDeletion = false;
+                    foreach (var blacklistTag in type.BlacklistedTags)
+                    {
+                        if (_tagSystem.HasTag(entity, blacklistTag))
+                        {
+                            //Logger.Info($"Skipped deletion for {ToPrettyString(entity)} due to blacklisted tag: {blacklistTag}");
+                            skipDeletion = true;
+                            break;
+                        }
+                    }
+                        if (skipDeletion)
+                        continue;
+                }
 
+                if (type.DeleteTags.Count > 0 && 
+                    _tagQuery.TryGetComponent(entity, out var tags))
+                {
+                    foreach (var tag in type.DeleteTags)
+                    {
+                        if (_tagSystem.HasTag(entity, tag) && 
+                            !_processedForDeletion.Contains(entity) && 
+                            !_entMan.IsQueuedForDeletion(entity) && 
+                            !_container.IsEntityInContainer(entity))
+                        {
+                            //Logger.Info($"Deleting: {ToPrettyString(entity)} | Type: {type.ID} | Tag: {tag}");
+                            _processedForDeletion.Add(entity);
+                            _entMan.QueueDeleteEntity(entity);
+                            break;
+                        }
+                    }
+                }
+                if (!_entMan.Deleted(entity) && !_entMan.IsQueuedForDeletion(entity) && !_processedForDeletion.Contains(entity))
+                {
+                    // TODO EXPLOSIONS turn explosions into entities, and pass the the entity in as the damage origin.
+                    _damageableSystem.TryChangeDamage(entity, damage * _damageableSystem.UniversalExplosionDamageModifier, ignoreResistances: true);
+                }
```

Comment:
Checking for tags and container is really expensive, plus you're not even re-using the components here.

Using damage thresholds is significantly more convenient code-wise and performant just tooling it is bad.